pip/req/req_set.py

```diff
@@ -253,6 +253,10 @@ def add_requirement(self, install_req, parent_req_name=None):
                     # scanning.
                     result = []
                 elif not install_req.constraint:
+                    if install_req.editable:
```

Comment:
Looks like you're meaning install_req.link, which has install_req.link.url, but looking at InstallRequrement, it looks like we don't need to take it all the way out to install_req.link.url.